src/jit/lsra.cpp

```diff
@@ -6653,23 +6656,25 @@ LinearScan::allocateRegisters()
                 }
                 else
 #endif // FEATURE_SIMD
-                if (currentRefPosition->IsActualRef())
+                if (currentRefPosition->RequiresRegister() ||
+                    currentRefPosition->IsRegOptional())
                 {
+                    // Though Lower/Codegen has indicated that it can generate code even if
+                    // no reg is allocated to this ref position, we will make an attempt
+                    // to get a busy reg if it is allocated to a lesser important ref position.
```

Comment:
OK, I see. However, I think this is still too confusing - it is not at all clear what `allocateReg` means. I've just realized that you still have `IsRegOptional()` returning false for the RefTypes that don't require a register (e.g. `RefTypeExpUse`). That was one of my fundamental concerns with the previous set of changes. If a `RefPosition` doesn't require a register, then `IsRegOptional()` should return true. Then there can be other analysis to determine which `RefPosition`s should be considered for allocating at the expense of spilling something else. We may eventually want to do that for some of the other RefTypes.